lib/modules/manager/devbox/schema.ts

```diff
@@ -0,0 +1,11 @@
+import { z } from 'zod';
+import { Jsonc } from '../../../util/schema-utils';
+
+export const DevboxFile = Jsonc.pipe(
+  z.object({
+    packages: z.union([
+      z.array(z.string()),
+      z.record(z.union([z.string(), z.object({ version: z.string() })])),
```

Comment:
@zharinov would a transformer simplify processing later? so we transform the record to array or the array to a record here?

Reply:
That sounds like it might simplify things, happy to implement that if you decide that's a cleaner way of doing this.

Reply:
Should I go ahead with doing that, or are we waiting on a response from @zharinov ?

Reply:
Yes, for example `z.object({ version: z.string() }).transform(({ version }) => version)` will simplify the inner union, then array and record could be transformed to array of deps, making the outer union to produce uniform type.

Ultimately, there should be nothing to transform after `safeParse`. Mell, maybe only retain `deps` length check, as I believe it must be refactored on our side to be performed outside `extractPackageFile()` call.